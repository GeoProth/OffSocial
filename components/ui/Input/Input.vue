<template>
  <input
    class="ui-input"
    :type="type"
    :value="value"
    :placeholder="holder || ''"
    @input="$emit('input', $event.target.value)"
    @keyup.enter="$emit('enter', $event.target.value)"
  >
</template>
<script>
export default {
  props: {
    value: [Number, String],
    type: String,
    holder: String
  },
  data: () => ({
    valid: false
  }),
  methods: {
  }
};
</script>
<style lang="stylus" scoped>
.ui-input {
  outline: none
  margin:10px;
  box-sizing: border-box;
  border:1px solid rgba(0,0,0,0.05)
  padding:10px;
  width:200px;
  border-radius:4px;
  font-size:15px;
  box-shadow: 0px 0px 10px #ccc;
}
.invalid {
  border-color: red;
  background: lightpink;
}
</style>
