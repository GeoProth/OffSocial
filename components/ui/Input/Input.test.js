import Input from "./Input";
describe("Input", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(Input);
  });
  test("renders", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("Emits an input event when typed into", async () => {
    wrapper.find(".ui-input").setValue("Hello");
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted().input.length).toBe(1);
  });
});
