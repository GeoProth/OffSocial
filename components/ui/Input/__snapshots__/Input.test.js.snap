// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders 1`] = `<input placeholder="" class="ui-input">`;
